nomos-services/data-availability/dispersal/src/adapters/mempool/kzgrs.rs

```diff
@@ -0,0 +1,115 @@
+// std
+use std::hash::Hash;
+use std::{fmt::Debug, marker::PhantomData};
+// crates
+use kzgrs_backend::dispersal::{self, BlobInfo};
+use nomos_core::da::blob::info::DispersedBlobInfo;
+use nomos_core::header::HeaderId;
+use nomos_da_sampling::backend::DaSamplingServiceBackend;
+use nomos_mempool::{
+    backend::MemPool, network::NetworkAdapter as MempoolAdapter, DaMempoolService, MempoolMsg,
+};
+use overwatch_rs::{
+    services::{relay::OutboundRelay, ServiceData},
+    DynError,
+};
+use rand::{RngCore, SeedableRng};
+use tokio::sync::oneshot;
+// internal
+use super::DaMempoolAdapter;
+
+type MempoolRelay<Payload, Item, Key> = OutboundRelay<MempoolMsg<HeaderId, Payload, Item, Key>>;
+
+pub struct KzgrsMempoolAdapter<
+    DaPoolAdapter,
+    DaPool,
+    SamplingBackend,
+    SamplingNetworkAdapter,
+    SamplingRng,
+    SamplingStorage,
+> where
+    DaPool: MemPool<BlockId = HeaderId>,
+    DaPoolAdapter: MempoolAdapter<Key = DaPool::Key>,
+    DaPoolAdapter::Payload: DispersedBlobInfo + Into<DaPool::Item> + Debug,
+    DaPool::Item: Clone + Eq + Hash + Debug + 'static,
+    DaPool::Key: Debug + 'static,
+{
+    pub mempool_relay: MempoolRelay<DaPoolAdapter::Payload, DaPool::Item, DaPool::Key>,
+    _phantom: PhantomData<(
+        SamplingBackend,
+        SamplingNetworkAdapter,
+        SamplingRng,
+        SamplingStorage,
+    )>,
+}
+
+#[async_trait::async_trait]
+impl<
+        DaPoolAdapter,
+        DaPool,
+        SamplingBackend,
+        SamplingNetworkAdapter,
+        SamplingRng,
+        SamplingStorage,
+    > DaMempoolAdapter
+    for KzgrsMempoolAdapter<
+        DaPoolAdapter,
+        DaPool,
+        SamplingBackend,
+        SamplingNetworkAdapter,
+        SamplingRng,
+        SamplingStorage,
+    >
+where
+    DaPool: MemPool<BlockId = HeaderId, Key = [u8; 32]>,
```

Comment:
that `[u8; 32]` is probably telling us that we need at least a type alias.
BTW, Why do we have a `BlockId` in the da mempool? I forgot, but may it be a leftover from the certificate?

Reply:
Agree on type alias, in this case it needs to be from Kzgrs backend (blobId), will add if not existing already. BlockId is not related to certificate afaik, will double check.

Reply:
`BlockId` is used for marking in block and other operations related to mempool backend.